src/cache.ts

```diff
@@ -0,0 +1,33 @@
+import * as queryString from 'query-string';
+
+import RequestOptions from './request-options';
+import Response from './response';
+
+export interface Cache {
+    read<T>(url: string, options: RequestOptions): Response<T> | null;
+    write<T>(url: string, options: RequestOptions, response: Response<T>): void;
+}
+
+export default class DefaultCache implements Cache {
+    private _cache = new Map<string, Response<any>>();
```

Comment:
curious what are the benefits of using a `Map` vs a regular object in this case? why did you opt for a Map?

Reply:
I would probably just use a regular object so we don't need to polyfill it for IE11

Reply:
@rtalvarez `Map` seems like the right data structure for this, no real benefit in this specific case vs a regular object other than showing intent on how it's supposed to be used. However, I changed it to a regular object to avoid polyfilling it as @davidchin mentioned.